packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -425,7 +425,17 @@ export class AngularWebpackPlugin {
       for (const sourceFile of builder.getSourceFiles()) {
         // Collect Angular template diagnostics
         if (!ignoreForDiagnostics.has(sourceFile)) {
-          diagnosticsReporter(angularCompiler.getDiagnostics(sourceFile));
+          // The below check should be removed once support for compiler 11.0 is dropped.
+          // Also, the below require should be changed to an ES6 import.
+          if (angularCompiler.getDiagnosticsForFile) {
+            // @angular/compiler-cli 11.1+
+            const { OptimizeFor } = require('@angular/compiler-cli/src/ngtsc/typecheck/api');
+            diagnosticsReporter(angularCompiler.getDiagnosticsForFile(sourceFile, OptimizeFor.WholeProgram));
+          } else {
+            // @angular/compiler-cli 11.0+
+            const getDiagnostics = angularCompiler.getDiagnostics as (sourceFile: ts.SourceFile) => ts.Diagnostic[];
```

Comment:
This will either need a `bind` here or a use of `call` at the invocation on the next line to maintain the object instance reference.

Reply:
You are right, good catch.